### PR TITLE
PCP-2154

### PR DESCRIPTION
**Pull request by wcrum**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-aws/pull/867#" title="Author self-approved">wcrum</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign  after the PR has been reviewed.
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-aws).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-aws/blob/spectro-release-4.1/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/cloud/services/iam/podidentitywebhook.go [hunk 1]
```diff
@@ -155,11 +155,38 @@ func reconcileDeployment(ctx context.Context, ns string, secret *corev1.Secret,
...
+		}
+	}
+
+	toleration := []corev1.Toleration{
+		{
+			Effect:   corev1.TaintEffect("NoSchedule"),
+			Key:      "node-role.kubernetes.io/control-plane",
+			Operator: "Exists",
+		},
+	}
```
Comment (pavansokkenagaraj) [inline pkg/cloud/services/iam/podidentitywebhook.go:173]:
```suggestion
	cpNoScheduleToleration := corev1.Toleration{
		Effect:   corev1.TaintEffect("NoSchedule"),
		Key:      "node-role.kubernetes.io/control-plane",
		Operator: "Exists",
	}

	// Check if the toleration already exists
	tolerationExists := false
	for _, t := range check.Spec.Template.Spec.Tolerations {
		if t.Key == cpNoScheduleToleration.Key && t.Operator == cpNoScheduleToleration.Operator && t.Effect == cpNoScheduleToleration.Effect {
			tolerationExists = true
			break
		}
	}
```

Comment (wcrum) [reply pkg/cloud/services/iam/podidentitywebhook.go:173]: @deepaks34 Do we want to include a new import?

Comment (pavansokkenagaraj) [reply pkg/cloud/services/iam/podidentitywebhook.go:173]: it is part of the go mod and used in code as a standard, so I would say yes

**Review by sadysnaat — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-aws/pull/867#pullrequestreview-1753406714" title="Approved">sadysnaat</a>*, *<a href="https://github.com/spectrocloud/cluster-api-provider-aws/pull/867#" title="Author self-approved">wcrum</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign  after the PR has been reviewed.
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-aws).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-aws/blob/spectro-release-4.1/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**